tox/__init__.py

```diff
@@ -34,6 +34,22 @@ class InterpreterNotFound(Error):
 
     class InvocationError(Error):
         """ an error while invoking a script. """
+        def __init__(self, *args):
+            super(exception.Error, self).__init__(*args)
+            self.command = args[0]
+            try:
+                self.exitcode = args[1]
+            except IndexError:
+                self.exitcode = None
+
+        def __str__(self):
+            str_ = "%s for command %s" % (self.__class__.__name__, self.command)
+            if self.exitcode:
+                str_ += " (exited with code %d)" % (self.exitcode)
+                if self.exitcode > 128:
+                    str_ += ("\nNote: On unix systems, an exit code larger than 128 "
+                             "often means a fatal error (e.g. 139=128+11: segmentation fault)")
```

Comment:
>128 means it exited due to a fatal signal

You can see the various signals that might trigger this in the table at the bottom of [this](https://en.wikipedia.org/wiki/Signal_(IPC)#POSIX_signals)

We could also compute this and show it to the user:

```console
$ python -c 'import time; time.sleep(100)'
^\Quit (core dumped)
$ echo $?
131
$ python -c '
> import signal
> sigs = {n: s for s, n in vars(signal).items() if s.startswith("SIG")}
> print(sigs[131 - 128])
> '
SIGQUIT
```

A `KeyError` on `sigs` (in my example) means it's probably not a fatal signal that killed it (also note that although `^C` sends SIGINT -- python sets a signal handler for this and reraises it as `KeyboardInterrupt` leading to an exit code of `1` instead of the expected `130`)

Reply:
Thanks for the link, added to the documentation that was in preparation.
Giving the signal value/name could be nice, since python uses directly the relevant `signals.h`; got to think about it.

Reply:
For reference: subprocess actually returns negative exit codes for signals.
See https://github.com/tox-dev/tox/pull/1401 for a followup/fix.